### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-navigation-ios#5]
The library should have a documentation comment for every public symbol. Swift uses reStructuredText format for documentation comments.

Additionally, we can look into generating HTML documentation based on these comments. The mobile team is quite happy with [jazzy](https://github.com/realm/jazzy), which relies on the compiler for parsing, so you know it’ll work with whatever syntax you throw at it. It also generates a Dash docset for those who like their documentation that way.

/cc @bsudekum @friedbunny

**Comment by 1ec5** [comment]
Xcode- and jazzy-compatible documentation was added in #250, unblocking this issue. Now you can run the following command line invocation:

```sh
jazzy --sdk iphonesimulator --github-file-prefix 'https://github.com/mapbox/mapbox-navigation-ios/tree/master' --module-version 0.4.0 --readme README.md --documentation=docs/*.md --root-url 'https://www.mapbox.com/navigation-sdk/ios/0.4.0/' --umbrella_header MapboxCoreNavigation/MapboxCoreNavigation.h --framework_root MapboxCoreNavigation --output ~/Desktop/docs/
```

to get Core Navigation documentation locally:

![jazzy](https://user-images.githubusercontent.com/1231218/27094815-5492aede-5020-11e7-9611-d7a027a4b25f.png)

Some caveats:

* We have two modules, MapboxCoreNavigation and MapboxNavigation. jazzy can only document one module per docset and lacks the ability to tie them together (or even cross-link symbols). I think this should be OK as long as our setup instructions are sufficiently fleshed out.
* The image in the “Storyboards” guide is broken.

We’ll need to find a good online home for this documentation. We could set up GitHub Pages temporarily, but hosting the docsets on mapbox.com would be preferable.

**Comment by friedbunny** [comment]
The Mapbox theme [lives here](https://github.com/mapbox/mapbox-gl-native/tree/9d276f3b0f0e5f1b5089a6cd727927361e6634ac/platform/darwin/docs/theme).

**Comment by ericrwolfe** [comment]
Closed via #285.
